Sonarqube 9.x support

#### Is your feature request related to a problem? Please describe.

> Yes, Sonar 9.x is not supported for report generation.
Detected SonarQube version: 9.0.1
[ERROR] SonarQube instance is not supported by cnesreport.

#### Describe the solution you'd like

> Add support for Sonar 9.x

#### Describe alternatives you've considered

> Nope :(

The /extension/cnesreport/report page works fine, but I get a 500 error when I try to generate a report (GET api/cnesreport/report?key=...)
`#web.log
Caused by: java.lang.NoClassDefFoundError: Could not initialize class fr.cnes.sonar.report.ReportCommandLine`

--- a/.github/workflows/java-continuous-integration.yml
+++ b/.github/workflows/java-continuous-integration.yml
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [ '8', '11' ]
+        java: [ '11' ]
     name: Java ${{ matrix.Java }} CI
     steps:
       - name: Check out repository code
@@ -43,7 +43,7 @@ jobs:
         run: mvn -B clean package
       - name: Test cnes-report
         env:
-          SONARQUBE_VERSION: 8.9-community
+          SONARQUBE_VERSION: 9.9-community
         run: |
           version=$(cat pom.xml | grep "<version>.*</version>" | head -1 |awk -F'[><]' '{print $3}');
           echo "Starting docker";

--- a/README.md
+++ b/README.md
@@ -72,7 +72,7 @@ java -jar sonar-cnes-report.jar -p projectId
 If you have installed cnes-report in your sonarqube: open web interface, click on "CNES Report" then choose a project.
 
 ##### Advanced usage (standalone)
-If you are using a secured instance of SonarQube, you can provide a SonarQube authentication token thanks to `-t` option and specify the url of the SonarQube instance with `-s`. The internal template for the text report can be replaced by the one given through `-r` option.
+If you are using a secured instance of SonarQube, you can provide a SonarQube user token (https://docs.sonarqube.org/latest/user-guide/user-account/generating-and-using-tokens/) thanks to `-t` option and specify the url of the SonarQube instance with `-s`. The internal template for the text report can be replaced by the one given through `-r` option.
 ````
 java -jar sonar-cnes-report.jar -t xuixg5hub345xbefu -s https://example.org:9000 -p projectId -r ./template.docx
 ````

--- a/README.md
+++ b/README.md
@@ -16,7 +16,7 @@ This tool can be used in standalone as a JAR executable (with the command line) 
 - Run an analysis with sonar-scanner, maven, gradle, msbuild, etc.
 - Execute cnesreport:
    - In standalone, thanks to command line
-   - Give your jar 777 permissions (chmod 777 `myfile.jar`)
+   - Give your jar read permissions for others (chmod o+r `myfile.jar`)
    - In plugin mode, copy jar in `/opt/sonarqube/extensions/plugins`, restart sonarqube, then click on "More" > "CNES Report".
 
 #### Installation
@@ -24,7 +24,7 @@ This tool can be used in standalone as a JAR executable (with the command line) 
 **cnesreport** does not need any installation. It is a portable Java application you can copy and run according to following examples. The only requirement is an **up-to-date JRE (>=1.8)**.
 
 ##### Plugin mode (Since 2.2.0)
-- Copy the sonar-cnes-report.jar with 777 rights in the plugin folder of sonarqube (On linux path should be like `/opt/sonarqube/extensions/plugins`)
+- Copy the sonar-cnes-report.jar with read rights for others in the plugin folder of sonarqube (On linux path should be like `/opt/sonarqube/extensions/plugins`)
 - Restart sonarqube (On linux: `sudo service sonar restart`)
 
 #### Get help

--- a/README.md
+++ b/README.md
@@ -16,6 +16,7 @@ This tool can be used in standalone as a JAR executable (with the command line) 
 - Run an analysis with sonar-scanner, maven, gradle, msbuild, etc.
 - Execute cnesreport:
    - In standalone, thanks to command line
+   - Give your jar 777 permissions (chmod 777 `myfile.jar`)
    - In plugin mode, copy jar in `/opt/sonarqube/extensions/plugins`, restart sonarqube, then click on "More" > "CNES Report".
 
 #### Installation
@@ -23,7 +24,7 @@ This tool can be used in standalone as a JAR executable (with the command line) 
 **cnesreport** does not need any installation. It is a portable Java application you can copy and run according to following examples. The only requirement is an **up-to-date JRE (>=1.8)**.
 
 ##### Plugin mode (Since 2.2.0)
-- Copy the sonar-cnes-report.jar in the plugin folder of sonarqube (On linux path should be like `/opt/sonarqube/extensions/plugins`)
+- Copy the sonar-cnes-report.jar with 777 rights in the plugin folder of sonarqube (On linux path should be like `/opt/sonarqube/extensions/plugins`)
 - Restart sonarqube (On linux: `sudo service sonar restart`)
 
 #### Get help
@@ -115,29 +116,42 @@ For legacy versions, check the wiki page here : [Note on legacy versions](https:
 <table>
  <tr>
   <td><b>cnesreport <br>\<br> SonarQube</b></td>
-  <td><b>3.0.x<br/>Standalone + Plugin</b></b></td>
-  <td><b>3.1.0<br/>Standalone + Plugin</b></b></td>
-  <td><b>3.2.x<br/>Standalone + Plugin</b></b></td>
-  <td><b>3.3.x<br/>Standalone + Plugin</b></b></td>
-  <td><b>4.0.0<br/>Standalone + Plugin</b></b></td>
-  <td><b>4.1.x<br/>Standalone + Plugin</b></b></td>
+  <td><b>3.0.x<br/>Standalone + Plugin</td>
+  <td><b>3.1.0<br/>Standalone + Plugin</td>
+  <td><b>3.2.x<br/>Standalone + Plugin</td>
+  <td><b>3.3.x<br/>Standalone + Plugin</td>
+  <td><b>4.0.0<br/>Standalone + Plugin</td>
+  <td><b>4.1.x<br/>Standalone + Plugin</td>
+  <td><b>4.2.x<br/>Standalone + Plugin</td>
  </tr>
  <tr>
-  <td><b>7.9.x (former LTS)</b></td>
+  <td><b>7.9.x</b></td>
   <td>X</td>
   <td>X</td>
   <td>X</td>
   <td>X</td>
+  <td>-</td>
   <td>-</td>
   <td>-</td>
  </tr>
  <tr>
-  <td><b>8.9.x (LTS)</b></td>
+  <td><b>8.9.x</b></td>
   <td>-</td>
   <td>-</td>
   <td>-</td>
   <td>-</td>
   <td>X</td>
+  <td>X</td>
+  <td>-</td>
+ </tr>
+ <tr>
+  <td><b>9.9.x (LTS)</b></td>
+  <td>-</td>
+  <td>-</td>
+  <td>-</td>
+  <td>-</td>
+  <td>-</td>
+  <td>-</td>
   <td>X</td>
  </tr>
 </table>

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "sonar-cnes-report-plugin",
     "license": "GPL-3.0",
-    "version": "4.1.2",
+    "version": "4.2.0",
     "devDependencies": {
         "@babel/core": "^7.8.7",
         "@babel/plugin-proposal-class-properties": "^7.8.3",

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>fr.cnes.sonar</groupId>
     <artifactId>cnesreport</artifactId>
-    <version>4.2</version>
+    <version>4.2.0</version>
     <packaging>sonar-plugin</packaging>
 
     <name>SonarQube CNES Report</name>
@@ -34,13 +34,14 @@
         <maven.compiler.target>11</maven.compiler.target>
         <junit.version>5.9.2</junit.version>
         <jacoco.version>0.8.8</jacoco.version>
-        <sonar.apiVersion>9.15.0.435</sonar.apiVersion>
+        <sonar.apiVersion>9.8.0.203</sonar.apiVersion>
+        <sonar.testingHarnessVersion>9.5.0.56709</sonar.testingHarnessVersion>
         <sonar.Version>9.9.0.65466</sonar.Version>
         <sonar-packaging-maven-plugin.version>1.21.0.505</sonar-packaging-maven-plugin.version>
         <sonar.pluginKey>cnesreport</sonar.pluginKey>
         <sonar.pluginUrl>https://github.com/cnescatlab/sonar-cnes-report</sonar.pluginUrl>
         <sonar.pluginOrganizationName>CNES</sonar.pluginOrganizationName>
-        <sonar.sources>src/main/java</sonar.sources>
+        <sonar.sources>src/main/java, src/main/js</sonar.sources>
         <sonar.test>src/test/java</sonar.test>
         <sonar.skipDependenciesPackaging>true</sonar.skipDependenciesPackaging>
         <sonar.pluginClass>fr.cnes.sonar.plugin.ReportSonarPlugin</sonar.pluginClass>
@@ -67,6 +68,10 @@
         <developer>
             <id>louisjdmartin</id>
             <name>Louis Martin</name>
+        </developer>
+        <developer>
+            <id>rodrigd</id>
+            <name>Diego Rodriguez</name>
         </developer>
         <developer>
             <id>ErwanGauduchon</id>
@@ -186,7 +191,7 @@
         <dependency>
             <groupId>org.sonarsource.sonarqube</groupId>
             <artifactId>sonar-testing-harness</artifactId>
-            <version>${sonar.Version}</version>
+            <version>${sonar.testingHarnessVersion}</version>
             <scope>test</scope>
         </dependency>
         <dependency>
@@ -347,7 +352,7 @@
             <plugin>
                 <groupId>com.github.eirslett</groupId>
                 <artifactId>frontend-maven-plugin</artifactId>
-                <version>1.12.1</version>
+                <version>1.9.1</version>
                 <executions>
                 <execution>
                     <phase>generate-resources</phase>
@@ -356,8 +361,8 @@
                     <goal>install-node-and-yarn</goal>
                     </goals>
                     <configuration>
-                        <nodeVersion>v14.16.1</nodeVersion>
-                        <yarnVersion>v1.22.10</yarnVersion>
+                        <nodeVersion>v16.14.0</nodeVersion>
+                        <yarnVersion>v1.22.5</yarnVersion>
                     </configuration>
                 </execution>
                 <execution>

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>fr.cnes.sonar</groupId>
     <artifactId>cnesreport</artifactId>
-    <version>4.1.3</version>
+    <version>4.2</version>
     <packaging>sonar-plugin</packaging>
 
     <name>SonarQube CNES Report</name>
@@ -29,13 +29,14 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <jdk.min.version>1.8</jdk.min.version>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
-        <junit.version>4.13.2</junit.version>
-        <jacoco.version>0.8.7</jacoco.version>
-        <sonar.apiVersion>8.9.0.43852</sonar.apiVersion>
-        <sonar-packaging-maven-plugin.version>1.20.0.405</sonar-packaging-maven-plugin.version>
+        <jdk.min.version>11</jdk.min.version>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <junit.version>5.9.2</junit.version>
+        <jacoco.version>0.8.8</jacoco.version>
+        <sonar.apiVersion>9.15.0.435</sonar.apiVersion>
+        <sonar.Version>9.9.0.65466</sonar.Version>
+        <sonar-packaging-maven-plugin.version>1.21.0.505</sonar-packaging-maven-plugin.version>
         <sonar.pluginKey>cnesreport</sonar.pluginKey>
         <sonar.pluginUrl>https://github.com/cnescatlab/sonar-cnes-report</sonar.pluginUrl>
         <sonar.pluginOrganizationName>CNES</sonar.pluginOrganizationName>
@@ -77,7 +78,7 @@
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-csv</artifactId>
-            <version>1.8</version>
+            <version>1.10.0</version>
             <scope>compile</scope>
         </dependency>
         <dependency>
@@ -87,7 +88,7 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>org.sonarsource.sonarqube</groupId>
+            <groupId>org.sonarsource.api.plugin</groupId>
             <artifactId>sonar-plugin-api</artifactId>
             <version>${sonar.apiVersion}</version>
             <scope>provided</scope>
@@ -95,19 +96,19 @@
         <dependency>
             <groupId>org.sonarsource.sonarqube</groupId>
             <artifactId>sonar-ws</artifactId>
-            <version>${sonar.apiVersion}</version>
+            <version>${sonar.Version}</version>
             <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>org.sonarsource.sonarqube</groupId>
             <artifactId>sonar-core</artifactId>
-            <version>${sonar.apiVersion}</version>
+            <version>${sonar.Version}</version>
         </dependency>
         <dependency>
             <!-- network use -->
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpclient</artifactId>
-            <version>4.5.13</version>
+            <version>4.5.14</version>
             <scope>compile</scope>
         </dependency>
         <dependency>
@@ -147,13 +148,13 @@
             <!-- https://mvnrepository.com/artifact/com.google.guava/guava -->
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>30.1.1-jre</version>
+            <version>31.1-jre</version>
         </dependency>
         <dependency>
             <!-- json parsing -->
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.8.9</version>
+            <version>2.10.1</version>
             <scope>compile</scope>
         </dependency>
         <dependency>
@@ -185,15 +186,20 @@
         <dependency>
             <groupId>org.sonarsource.sonarqube</groupId>
             <artifactId>sonar-testing-harness</artifactId>
-            <version>${sonar.apiVersion}</version>
+            <version>${sonar.Version}</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-api</artifactId>
-            <version>5.7.2</version>
+            <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+	      <groupId>org.junit.vintage</groupId>
+	      <artifactId>junit-vintage-engine</artifactId>
+	      <version>${junit.version}</version>
+	    </dependency>
         <dependency>
             <groupId>com.github.stefanbirkner</groupId>
             <artifactId>system-rules</artifactId>
@@ -203,7 +209,7 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>3.12.4</version>
+            <version>5.1.1</version>
             <scope>test</scope>
         </dependency>
 
@@ -234,7 +240,7 @@
 
             <plugin>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.2.4</version>
+                <version>3.4.1</version>
                 <executions>
                     <execution>
                         <phase>package</phase>
@@ -268,7 +274,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.8.1</version>
+                <version>3.10.1</version>
                 <configuration>
                     <source>${jdk.min.version}</source>
                     <target>${jdk.min.version}</target>
@@ -276,24 +282,23 @@
             </plugin>
 
             <plugin>
-                <!-- UTF-8 bundles are not supported by Java, so they must be converted during build -->
-                <groupId>org.codehaus.mojo</groupId>
-                <artifactId>native2ascii-maven-plugin</artifactId>
-                <version>1.0-beta-1</version>
-                <executions>
-                    <execution>
-                        <?m2e execute onConfiguration?>
-                        <goals>
-                            <goal>native2ascii</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
+	        	<!-- UTF-8 bundles are not supported by Java, so they must be converted during build -->
+	        	<groupId>org.codehaus.mojo</groupId>
+	        	<artifactId>native2ascii-maven-plugin</artifactId>
+	        	<version>2.0.1</version>
+	        	<executions>
+	          		<execution>
+	            		<goals>
+	              			<goal>resources</goal>
+	            		</goals>
+	          		</execution>
+	        	</executions>
+	      	</plugin>
 
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-project-info-reports-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.4.2</version>
                 <configuration>
                     <dependencyLocationsEnabled>false</dependencyLocationsEnabled>
                 </configuration>
@@ -302,14 +307,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.0.0-M5</version>
-                <dependencies>
-                    <dependency>
-                        <groupId>org.apache.maven.surefire</groupId>
-                        <artifactId>surefire-junit47</artifactId>
-                        <version>3.0.0-M5</version>
-                    </dependency>
-                </dependencies>
+                <version>3.0.0-M8</version>
             </plugin>
 
             <plugin>
@@ -349,7 +347,7 @@
             <plugin>
                 <groupId>com.github.eirslett</groupId>
                 <artifactId>frontend-maven-plugin</artifactId>
-                <version>1.11.3</version>
+                <version>1.12.1</version>
                 <executions>
                 <execution>
                     <phase>generate-resources</phase>
@@ -358,8 +356,8 @@
                     <goal>install-node-and-yarn</goal>
                     </goals>
                     <configuration>
-                    <nodeVersion>v14.16.1</nodeVersion>
-                    <yarnVersion>v1.22.10</yarnVersion>
+                        <nodeVersion>v14.16.1</nodeVersion>
+                        <yarnVersion>v1.22.10</yarnVersion>
                     </configuration>
                 </execution>
                 <execution>

--- a/pom.xml
+++ b/pom.xml
@@ -41,7 +41,7 @@
         <sonar.pluginKey>cnesreport</sonar.pluginKey>
         <sonar.pluginUrl>https://github.com/cnescatlab/sonar-cnes-report</sonar.pluginUrl>
         <sonar.pluginOrganizationName>CNES</sonar.pluginOrganizationName>
-        <sonar.sources>src/main/java, src/main/js</sonar.sources>
+        <sonar.sources>src/main/java</sonar.sources>
         <sonar.test>src/test/java</sonar.test>
         <sonar.skipDependenciesPackaging>true</sonar.skipDependenciesPackaging>
         <sonar.pluginClass>fr.cnes.sonar.plugin.ReportSonarPlugin</sonar.pluginClass>

--- a/src/main/java/fr/cnes/sonar/report/factory/ServerFactory.java
+++ b/src/main/java/fr/cnes/sonar/report/factory/ServerFactory.java
@@ -32,7 +32,7 @@ public class ServerFactory {
 
     /** List of SonarQube versions which are supported by cnesreport. */
     private static final List<String> SUPPORTED_VERSIONS = Arrays.asList(
-            "8.9", "8.9.*");
+            "8.9", "8.9.*", "9.9", "9.9.*");
 
     /** Url of the SonarQube server. */
     private String url;

--- a/src/main/js/common/api.js
+++ b/src/main/js/common/api.js
@@ -6,7 +6,7 @@ import { getJSON, postJSON, post } from "sonar-request";
 
 // Function used to get current SonarQube Server version
 export function isCompatible() {
-  const COMPATIBILITY_PATTERN = /8.9(?:\..*)?(?:-.*)?/;
+  const COMPATIBILITY_PATTERN = /[8-9]\.9(\.[0-9])*/;                 
 
   return getJSON("/api/system/status").then(response => {
     return response.version.match(COMPATIBILITY_PATTERN) != null;

--- a/src/main/js/conf/env.js
+++ b/src/main/js/conf/env.js
@@ -1,7 +1,8 @@
 /**
  * Script taken from https://github.com/SonarSource/sonar-custom-plugin-example
- * Used to handle environment variable
- */
+ * Grab NODE_ENV and REACT_APP_* environment variables and prepare them to be
+ * injected into the application via DefinePlugin in Webpack configuration.
+*/
 
 const REACT_APP = /^REACT_APP_/i;
 

--- a/src/test/ut/java/fr/cnes/sonar/report/CommonTest.java
+++ b/src/test/ut/java/fr/cnes/sonar/report/CommonTest.java
@@ -47,7 +47,7 @@ public abstract class CommonTest {
     /**
      * Branch name.
      */
-    protected static final String BRANCH = "master";
+    protected static final String BRANCH = "main";
     /**
      * Quality Gate name.
      */

